tokenizing-text-files.ipynb

```diff
@@ -0,0 +1,462 @@
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for term, count in first_doc[\"trigramCount\"].most_common(25):\n",
+    "    print(term, count)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Print the first 500 characters of \"page\" 20"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "print(first_doc[\"fullText\"][20][:500])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Generate a Constellate gzip file\n",
+    "\n",
+    "You may now want to create a gzip file so that it matches what you have downloaded from Constellate. You could also then use the `gzip_reader` that's part of the `tdm_client` to read it. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "output_file = text_file_directory + os.sep + \"sample_gutenberg_dataset.json.gzip\"\n",
+    "\n",
+    "with gzip.open(output_file, \"wb\") as handle:\n",
+    "    for doc in documents:\n",
+    "        # Convert the document to a string and add the line separator\n",
+    "        raw = json.dumps(doc) + \"\\n\"\n",
+    "        handle.write(raw.encode())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now use the dataset reader to read your file back in and verify it is what we expect."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from tdm_client import dataset_reader\n",
+    "\n",
+    "for doc in dataset_reader(output_file):\n",
+    "    print(doc[\"title\"], doc[\"creator\"], doc[\"publicationYear\"])\n",
+    "    # See note about assert\n",
+    "    assert(doc[\"unigramCount\"] is not None)\n",
+    "    assert(doc[\"fullText\"] is not None)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Notes\n",
+    "\n",
+    "* File paths - in Unix based systems (including Linux and MacOS), a file is separated with a `/`. On Windows the separator is a `\\`. Python includes the helpful `os.sep` to find the correct file separator for your system. This allows the notebook to run just fine on multiple operating systems.\n",
+    "\n",
+    "* Pagination - since we are working with plain text files from Project Gutenberg, we no longer have pagination. Here we are using a simple rule of thumb, if there are three consecutive line breaks, treat this as a page break. This is unlikelly to work well across all Project Gutenberg content but should be sufficeint for demonstration purposes. You may be curious in more sophisticated attempts to format Project Gutenberg books, such a [chapterize](https://github.com/JonathanReeve/chapterize) by [Jonathan Reeve](https://github.com/JonathanReeve).\n",
```

Comment:
```suggestion
    "* Pagination - since we are working with plain text files from Project Gutenberg, we no longer have pagination. Here we are using a simple rule of thumb: if there are three consecutive line breaks, treat this as a page break. This is unlikely to work well across all Project Gutenberg content but should be sufficient for demonstration purposes. You may be curious about more sophisticated attempts to format Project Gutenberg books, such a [chapterize](https://github.com/JonathanReeve/chapterize) by [Jonathan Reeve](https://github.com/JonathanReeve).\n",
```